examples/DotNetCore/ConsoleAppWithTableStorageReference/Program.cs

```diff
@@ -0,0 +1,182 @@
+            string myShopDisplayedColumns = Configuration.GetValue<string>("MyShop:DisplayedColumns");
+            string[] allowedProductFields = new string[] { "Product", "Quantity", "OnSale" };
+            string[] columnsToDisplay;
+
+            if (!string.IsNullOrEmpty(myShopDisplayedColumns))
+            {
+                columnsToDisplay = myShopDisplayedColumns.Split(' ');
+            }
+            else
+            {
+                columnsToDisplay = allowedProductFields;
+            }
+
+            sb.AppendLine("Your shop inventory:\n");
+            var tableFormat = string.Empty;
+
+            for (int i = 0; i < columnsToDisplay.Length; i++)
+            {
+                tableFormat += "{" + i + ", -20}";
+            }
+
+            sb.AppendFormat(tableFormat, columnsToDisplay);
+            sb.AppendLine();
+
+            foreach (Product product in tableContent)
+            {
+                string[] productValues = new string[] { product.Name, product.Quantity.ToString(), product.OnSale.ToString() };
+                string[] fieldsToPrint = new string[columnsToDisplay.Length];
+                int fieldsToPrintIndex = 0;
+
+                foreach (string field in allowedProductFields)
+                {
+                    if (columnsToDisplay.Contains(field))
+                    {
+                        fieldsToPrint[fieldsToPrintIndex] = productValues[Array.IndexOf(columnsToDisplay, field)];
+                        fieldsToPrintIndex++;
+                    }
+                }
+
+                sb.AppendFormat(tableFormat, fieldsToPrint);
+                sb.AppendLine();
+            }
+
+            return sb;
+        }
+
+        private static async Task<string> ReadTableContentAsync(Uri tableUri)
+        {
+            string[] pathSegments = tableUri.Segments;
+
+            // The last segment in the path should be the table name
+            string tableName = pathSegments[pathSegments.Length - 1];
+
+            var tableClient = new TableClient(tableUri, tableName, new DefaultAzureCredential());
+
+            // Adding example products to display
+            var prod1 = new Product()
+            {
+                RowKey = "68719518388",
+                PartitionKey = "gear-surf-surfboards",
+                Name = "Ocean Surfboard",
+                Quantity = 8,
+                OnSale = true
+            };
+
+            var prod2 = new Product()
+            {
+                RowKey = "68719518390",
+                PartitionKey = "gear-surf-surfboards",
+                Name = "Sand Surfboard",
+                Quantity = 5,
+                OnSale = false
+            };
+
+            await tableClient.UpsertEntityAsync(prod1);
+            await tableClient.UpsertEntityAsync(prod2);
+
+            var products = tableClient.Query<Product>();
+
+            return JsonSerializer.Serialize(products.ToList());
+        }
+
+        private static void Configure()
```

Comment:
Can you move this method to the top, right after main method? It just highlights our product logic before table manipulations.